### DISCUSSION TITLE
Reporting

### DISCUSSION THREAD

**Discussion by KaylenS** [stayonhire/feedback#30] [Ideas / Wishlist]
Hi Dylan,

Does StayOnHire have any reporting functionality? E.g. is there a way we can export a report for all active jobs, or completed jobs, for all dates or within a specified date range? If this is not a current function, could we please get this happening? Priority 7/10.

**Comment by dylanhru** [comment]
Hi Kaylen, 

I like this idea and yes I can start scoping up and building it. 

Do you have any example reports/templates that you could email me directly? Or even a screenshot (or redacted example of one you really like) from somewhere else?

I will be mainly looking for the exact metrics and measurements I need to export for it to be valuable to ABP as I don't want to assume anything. Speak soon and hopefully we can add it into one of our Feedback Fridays coming up! :)

**Comment by WadeRolfe** [comment]
Hi Dylan,

It seems we've started getting the daily notification emails with the task lists come through. This is a great addition and something we can use quite a bit moving forward.

Is this something we can get set up to do more personalised task emails sent out.

For instance: I set Kaylen a task for this Friday to off hire a generator. She would get an email with only the allocated tasks set for her and no one else gets that email.

This would allow everyone in the team to set the required reminders and plan for servicing, sales functions and admin.

Hope that makes sense?

**Reply by dylanhru** [reply]
Hi @WadeRolfe 
We configured the email reminders to send to all users who have any tasks assigned to them using the 'Staff Responsible' field.

In the example task you mentioned above, for @KaylenS to off hire and asset on Friday, technically if you also assigned a second person to the same task, they will also receive a daily email reminding them of the task.

This would also mean any other tasks you create and assign to other workers, they will also get emails each morning to review their tasks (which can be accessed from the mobile app).

Is this in line with what you're looking for?

**Comment by dylanhru** [comment]
Hi @WadeRolfe,

I think I know what you're talking about after managing my own To Dos and reminders in StayOnHire. @twrayden and I are going to try and squeeze this fix in today (or early Monday), to tweak the contents of these emails to only display the recipient's tasks (not company-wide tasks).

We're also going to cut down the branding and keep it simple, alongside with left-aligning content on mobile devices.

![Screenshot 2023-11-03 at 10 27 09 am](https://github.com/stayonhire/feedback/assets/34927957/94ea4112-78a1-4cdc-a62b-2ff5681850f2)

**Reply by dylanhru** [reply]
This has now been deployed @WadeRolfe. You should see the newer reminder come through, that is specific to each user's To Dos.

**Comment by WadeRolfe** [comment]
Outstanding team, looking forward to it.

Kind regards,

Wade Rolfe
AB Power Generation
0408288082
***@***.***

On 3 Nov 2023, at 9:19 pm, Dylan Hrurukvit ***@***.***> wrote:

﻿

This has now been deployed @WadeRolfe<https://github.com/WadeRolfe>. You should see the newer reminder come through, that is specific to each user's To Dos.

**Comment by WadeRolfe** [comment]
These notifications are coming through nicely team, thanks. I will start using them and allocating them for others in the team and see how they come through.

**Comment by dylanhru** [comment]
Toby (AFE) notes:

Reporting company performance as a whole
- client value / performance
- Machine Make value v performance
- Machine onhire total 
- service history performance

Company efficiency as a %

**Comment by dylanhru** [comment]
Additional scope details:

Analytics / Reporting - Visualise data from your StayOnHire account using utilisation, customer details, asset groups/categories, revenue and other metrics. We will provide 1-3 defaults analytics, though the intent would be that user's build their own analytics that are important to them -meaning customisation will be a major element of analytics. Data will be live and will be able to be downloaded or screenshotted but will not inherently produce an A4 'report'; as most users have different reporting needs.

![280898133-dd6f45e0-4ae6-46ca-9a51-b586531c530d](https://github.com/stayonhire/feedback/assets/34927957/edc2b225-5fd8-4f37-b492-341bcd5e2e32)
